lib/ballot.go

```diff
@@ -311,12 +326,21 @@ func (b *BallotBoxes) IsVoted(ballot Ballot) bool {
 	return vr.IsVoted(ballot)
 }
 
-func (b *BallotBoxes) AddVotingResult(vr *VotingResult, bb *BallotBox) (err error) {
+func (b *BallotBoxes) AddVotingResult(vr *VotingResult, ballot Ballot) (err error) {
 	b.Lock()
 	defer b.Unlock()
 
-	b.Results[vr.MessageHash] = vr
-	bb.AddVotingResult(vr) // TODO detect error
+	if ballot.CanFitInVotingBox() {
+		err = b.VotingBox.AddVotingResult(vr)
+	} else if ballot.CanFitInWaitingBox() {
+		err = b.WaitingBox.AddVotingResult(vr)
+	} else {
+		err = sebakerror.ErrorBallotHasInvalidState
+	}
+
+	if err == nil {
+		b.Results[vr.MessageHash] = vr
+	}
```

Comment:
Would you please check this part again? @spikeekips

Reply:
@Charleslee522 Good job and one more thing. The `error` in this part also can be `nil` in https://github.com/bosnet/sebak/pull/115/files/743e1a98dd2acf03cb2699c86510962870a5ec98#diff-2231e829007317368e86620b46b6bfcaR395